integration-tests/features/steps/common.py

```diff
@@ -978,6 +978,59 @@ def stack_analysis_check_companion_packages(context):
             % companion_package
 
 
+def get_user_components(json_data):
+    path = "result/0/user_stack_info/analyzed_dependencies"
+    return get_value_using_path(json_data, path)
+
+
+def get_alternate_components(json_data):
+    path = "result/0/recommendations/alternate"
+    return get_value_using_path(json_data, path)
+
+
+def check_attribute_presence(node, attribute_name):
+    assert attribute_name in node, \
+        "'%s' attribute is expected in the node, " \
+        "found: %s attributes " % (attribute_name, ", ".join(node.keys()))
+
+
+def perform_alternate_components_validation(json_data):
+    user_components = get_user_components(json_data)
+
+    # in order to use the 'in' operator later we need to have a sequence
+    # of dictionaries with 'name' and 'version' keys
+    user_components = [{"name": c["package"],
+                        "version": c["version"]} for c in user_components]
+    alternate_components = get_alternate_components(json_data)
+
+    for alternate_component in alternate_components:
+
+        check_attribute_presence(alternate_component, "name")
+        name = alternate_component.get("name")
```

Comment:
`name = alternate_component["name"]` since you're already sure it's there

Also, the `name` variable doesn't seem to be used anywhere after this assignment.